precompiles/erc20/errors.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package erc20
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/x/authz"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/crypto"
+	evmtypes "github.com/evmos/evmos/v15/x/evm/types"
+)
+
+var (
+	ErrInsufficientAllowance        = errors.New("ERC20: insufficient allowance")
+	ErrTransferAmountExceedsBalance = errors.New("ERC20: transfer amount exceeds balance")
+)
+
+// BuildExecRevertedErr returns a mocked error that should align with the
+// behavior of the original ERC20 Solidity implementation.
+//
+// FIXME: This is not yet producing the correct reason bytes.
+func BuildExecRevertedErr(reason string) (error, error) {
+	// The reason bytes are prefixed with this byte array -> see UnpackRevert in ABI implementation in Geth.
+	prefixBytes := crypto.Keccak256([]byte("Error(string)"))
```

Comment:
let's make this a variable instead outside the scope of this function, rename to `errorSignature`

Reply:
Done 👍